python/meterpreter/ext_server_stdapi.py

```diff
@@ -1216,6 +1217,85 @@ def stdapi_sys_config_getuid(request, response):
     response += tlv_pack(TLV_TYPE_USER_NAME, username)
     return ERROR_SUCCESS, response
 
+@register_function_if(has_windll)
+def stdapi_sys_config_getprivs(request, response):
+    TOKEN_QUERY = 0x0008
+    TOKEN_ADJUST_PRIVILEGES = 0x0020
+    SE_PRIVILEGE_ENABLED = 0x00000002
+
+    GetCurrentProcess = ctypes.windll.kernel32.GetCurrentProcess
+    GetCurrentProcess.restype = ctypes.c_void_p
+
+    advapi32 = ctypes.windll.advapi32
+    OpenProcessToken = advapi32.OpenProcessToken
+    OpenProcessToken.argtypes = [ctypes.c_void_p, ctypes.c_uint32, ctypes.POINTER(ctypes.c_void_p)]
+    OpenProcessToken.restype = ctypes.c_bool
+
+    LookupPrivilegeValue = advapi32.LookupPrivilegeValueW
+    LookupPrivilegeValue.argtypes = [ctypes.c_wchar_p, ctypes.c_wchar_p, ctypes.POINTER(LUID)]
+    LookupPrivilegeValue.restype = ctypes.c_bool
+
+    AdjustTokenPrivileges = advapi32.AdjustTokenPrivileges
+    AdjustTokenPrivileges.argtypes = [ctypes.c_void_p, ctypes.c_bool, PTOKEN_PRIVILEGES, ctypes.c_uint32, PTOKEN_PRIVILEGES, ctypes.POINTER(ctypes.c_uint32)]
+    AdjustTokenPrivileges.restype = ctypes.c_bool
+
+    token = ctypes.c_void_p()
+    success = OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, token)
+    if not success:
+        return error_result_windows(), response
+
+    priv_list = [
```

Comment:
It looks like this is missing some of the entries that the windows side will [enumerate](https://github.com/rapid7/metasploit-payloads/blob/261aaaace4ee5161d2fbbb4d291ce886d5eb1973/c/meterpreter/source/extensions/stdapi/server/sys/config/config.c#L286).

The values can be looked up from the docs [here](https://learn.microsoft.com/en-us/windows/win32/secauthz/privilege-constants).

The Windows one is missing from from the MS docs. We probably want to add all the permissions that are documented in alphabetical order (as they are in the docs) then I can update the Windows Meterpreter to add the missing privileges there.

Reply:
On looking at that list more closely, I'm not sure that's the list of privileges we'd want to go with. I tested actually requesting all of those and the `SeDelegateSessionUserImpersonatePrivilege` was the only one my session was able to active when running as SYSTEM.

From looking at the docs you linked and taking `SeDenyNetworkLogonRight` as an example:

> [Deny access to this computer from the network](https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/deny-access-to-this-computer-from-the-network)

I don't think we'd want to deny anything.

In comparison with the link I shared references usage with `LookupPrivilegeValue` which is what we're doing here. It also has `SeDelegateSessionUserImpersonatePrivilege` as the only privilege that's missing from the existing list used by Windows and was one I was able to confirm is able to be obtained once added.

Lets just add `SeDelegateSessionUserImpersonatePrivilege` to the Windows Meterpreter, then use the same list from there.

Reply:
Sounds good to me and yeah wasn't sure about the Deny ones, I listed those before I realized the code was trying to enable those privileges as well vs just enumerate them.

Reply:
Fixed with https://github.com/rapid7/metasploit-payloads/pull/607/commits/ca738c412c1b0e76306e8d4e1015c5a571509bf8